.azure-pipelines/ci-build.yml

```diff
@@ -12,298 +12,253 @@ pr:
-      strategy:
-        runOnce:
-          deploy:
-            pool:
-              vmImage: ubuntu-latest
-            steps:
-            # Install the nuget tool.
-            - task: NuGetToolInstaller@0
-              displayName: 'Use NuGet >=5.2.0'
-              inputs:
-                versionSpec: '>=5.2.0'
-                checkLatest: true
-            - task: DownloadPipelineArtifact@2
-              displayName: Download nupkg from artifacts
-              inputs:
-                artifact: Nugets
-                source: current
-            - task: PowerShell@2
-              displayName: 'Extract release information to pipeline'
-              inputs:
-                targetType: 'filePath'
-                filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
-                pwsh: true
-                arguments: '-packageDirPath "$(Pipeline.Workspace)/Nugets/"'
-            - task: NuGetCommand@2
-              displayName: 'NuGet push'
-              inputs:
-                command: push
-                packagesToPush: '$(Pipeline.Workspace)/Nugets/Microsoft.Kiota.Abstractions.*.nupkg'
-                nuGetFeedType: external
-                publishFeedCredentials: 'Kiota Nuget Connection'
-            - task: GitHubRelease@1
-              displayName: 'GitHub release (create)'
-              inputs:
-                gitHubConnection: 'Kiota_Release'
-                target: $(Build.SourceVersion)
-                tagSource: userSpecifiedTag
-                tag: 'v$(VERSION_STRING)'
-                title: '$(VERSION_STRING)'
-                releaseNotesSource: inline
-                assets: '!**/**'
-                changeLogType: issueBased
-                isPreRelease : '$(IS_PRE_RELEASE)'
-                addChangeLog : true 
+    - stage: deploy
+      condition: and(contains(variables['build.sourceBranch'], 'refs/heads/main'), succeeded())
+      dependsOn: build
+      jobs:
+        - deployment: deploy_dotnet_abstractions
+          dependsOn: []
+          environment: nuget-org
+          strategy:
+            runOnce:
+              deploy:
+                pool:
+                  vmImage: ubuntu-latest
+                steps:
+                # Install the nuget tool.
+                - task: NuGetToolInstaller@0
+                  displayName: 'Use NuGet >=5.2.0'
+                  inputs:
+                    versionSpec: '>=5.2.0'
+                    checkLatest: true
+                - task: DownloadPipelineArtifact@2
+                  displayName: Download nupkg from artifacts
+                  inputs:
+                    artifact: Nugets
+                    source: current
+                - task: PowerShell@2
+                  displayName: 'Extract release information to pipeline'
+                  inputs:
+                    targetType: 'filePath'
+                    filePath: $(Pipeline.Workspace)\Nugets\scripts\GetNugetPackageVersion.ps1
```

Comment:
Any chance you can double check these paths? I suspect the script will fail given the `Nugets` folder may not exist